loky/backend/context.py

```diff
@@ -126,44 +83,37 @@ def cpu_count(only_physical_cores=False):
     any other way such as: process affinity, restricting CFS scheduler policy
     or the LOKY_MAX_CPU_COUNT environment variable. If the number of physical
     cores is not found, return the number of logical cores.
- 
+
     It is also always larger or equal to 1.
     """
-    # TODO: use os.cpu_count when dropping python 2 support
-    try:
-        cpu_count_mp = mp.cpu_count()
-    except NotImplementedError:
-        cpu_count_mp = 1
+    cpu_count_mp = os.cpu_count() or 1
 
     cpu_count_user = _cpu_count_user(cpu_count_mp)
     aggregate_cpu_count = min(cpu_count_mp, cpu_count_user)
 
-    if only_physical_cores:
-        cpu_count_physical, exception = _count_physical_cores()
-        if cpu_count_user < cpu_count_mp:
-            # Respect user setting
-            cpu_count = max(cpu_count_user, 1)
-        elif cpu_count_physical == "not found":
-            # Fallback to default behavior
-            if exception is not None:
-                # warns only the first time
-                warnings.warn(
-                    "Could not find the number of physical cores for the "
-                    "following reason:\n" + str(exception) + "\n"
-                    "Returning the number of logical cores instead. You can "
-                    "silence this warning by setting LOKY_MAX_CPU_COUNT to "
-                    "the number of cores you want to use.")
-                if sys.version_info >= (3, 5):
-                    # TODO remove the version check when dropping py2 support
-                    traceback.print_tb(exception.__traceback__)
-
-            cpu_count = max(aggregate_cpu_count, 1)
-        else:
-            return cpu_count_physical
-    else:
-        cpu_count = max(aggregate_cpu_count, 1)
+    if not only_physical_cores:
+        return max(aggregate_cpu_count, 1)
+
+    if cpu_count_user < cpu_count_mp:
+        # Respect user setting
+        return max(cpu_count_user, 1)
+
+    cpu_count_physical, exception = _count_physical_cores()
+    if cpu_count_physical != "not found":
+        return cpu_count_physical
```

Comment:
Maybe just return `max(min(cpu_count_user, cpu_count_physical), 1)` here.

Reply:
That wouldn't be enough because of the if condition on `cpu_count_user` and `cpu_count_mp` before.

Reply:
But if we drop condition on `cpu_count_user` and `cpu_count_mp`, and move directly to this one?

Reply:
Let's keep the original logic unchanged in this PR and open a dedicated PR if you want to improve this with a dedicated changelog entry to document the change (and with dedicated tests, if possible).